### PR TITLE
Pagination sorting swagger

### PR DESCRIPTION
**Pull request by vfilimo**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -94,6 +94,13 @@
 			<artifactId>hibernate-validator</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+			<version>2.3.0</version>
+		</dependency>
+
+
```
Comment (sarakhmen) [inline pom.xml]:
```suggestion
```

File: src/main/java/online/store/book/controller/BookController.java [hunk 2]
```diff
@@ -16,39 +19,47 @@
...
 @RequestMapping("/books")
 @RequiredArgsConstructor
 public class BookController {
     private final BookService bookService;
 
+    @Operation(summary = "Get all books", description = "Get all books")
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (sarakhmen) [inline src/main/java/online/store/book/controller/BookController.java]: please add default pageable values, fix in all places

**Review by sarakhmen — CHANGES_REQUESTED** [review]
good job, let's slightly improve your solution :)

File: src/main/java/online/store/book/controller/BookController.java [hunk 3]
```diff
@@ -16,39 +20,54 @@
...
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(
+            @PageableDefault(size = DEFAULT_PAGE_SIZE, page = DEFAULT_PAGE,
+                    sort = DEFAULT_SORT_PARAMETER) Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
+    @Operation(summary = "Get all books by search parameters",
+            description = "Get all books by tittle and author")
```
Comment (Elena-Bruyako) [inline src/main/java/online/store/book/controller/BookController.java]:
```suggestion
            description = "Get all books by title and author")
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
one spelling mistake

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**